app/src/ui/repositories-list/repository-list-item-context-menu.ts

```diff
@@ -0,0 +1,112 @@
+import { Repository } from '../../models/repository'
+import { IMenuItem } from '../../lib/menu-item'
+import { enableRepositoryAliases } from '../../lib/feature-flag'
+import { Repositoryish } from './group-repositories'
+import { clipboard } from 'electron'
+import {
+  RevealInFileManagerLabel,
+  DefaultEditorLabel,
+} from '../lib/context-menu'
+
+interface IRepositoryListItemContextMenuConfig {
+  repository: Repositoryish
+  shellLabel: string
+  externalEditorLabel: string | undefined
+  askForConfirmationOnRemoveRepository: boolean
+  onViewOnGitHub: (repository: Repositoryish) => void
+  onOpenInShell: (repository: Repositoryish) => void
+  onShowRepository: (repository: Repositoryish) => void
+  onOpenInExternalEditor: (repository: Repositoryish) => void
+  onRemoveRepository: (repository: Repositoryish) => void
+  onChangeRepositoryAlias: (repository: Repository) => void
+  onRemoveRepositoryAlias: (repository: Repository) => void
+}
+
+export const generateRepositoryListContextMenu = (
+  config: IRepositoryListItemContextMenuConfig
+) => {
+  const missing =
+    config.repository instanceof Repository && config.repository.missing
```

Comment:
I think it'd make sense to extract `repository` into a separate variable here to tidy things up a bit.

```suggestion
  const { repository } = config
  const missing =
    config.repository instanceof Repository && config.repository.missing
```